tracer/src/Datadog.Trace/AppSec/Coordinator/SecurityCoordinator.cs

```diff
@@ -104,6 +86,121 @@ internal readonly partial struct SecurityCoordinator
 
     internal static Span TryGetRoot(Span span) => span.Context.TraceContext?.RootSpan ?? span;
 
+    public IResult? RunWafForUser(string? userId = null, string? userLogin = null, string? userSessionId = null, bool fromSdk = false, Dictionary<string, string>? otherTags = null)
+    {
+        if (_httpTransport.IsBlocked)
+        {
+            return null;
+        }
+
+        IResult? result = null;
+        Dictionary<string, object>? addresses = null;
+        try
+        {
+            var additiveContext = GetOrCreateAdditiveContext();
+            if (additiveContext?.ShouldRunWith(_security, userId, userLogin, userSessionId, fromSdk) is { Count: > 0 } userAddresses)
+            {
+                addresses = userAddresses.ToDictionary(k => k.Key, object (v) => v.Value);
+
+                if (otherTags is not null)
+                {
+                    foreach (var kvp in otherTags)
+                    {
+                        if (!addresses.ContainsKey(kvp.Key))
+                        {
+                            addresses.Add(kvp.Key, kvp.Value);
+                        }
+                    }
+                }
+
+                SecurityReporter.LogAddressIfDebugEnabled(addresses);
+
+                // run the WAF and execute the results
+                result = additiveContext.Run(addresses, _security.Settings.WafTimeoutMicroSeconds);
```

Comment:
I guess that this is modifying the session fingerprints, right? I have not seen any test related to session fingerprints, though. Should we test in this PR that the fingerprints are generated correctly?

Reply:
I added it in the snapshots [here](https://github.com/DataDog/dd-trace-dotnet/blob/a064bf56ef3b9acaccf1d246d43501de1f701faa/tracer/test/snapshots/Security.AspNetCore5AutoUserEvents.SecurityOn.identmode-TestAuthenticatedRequest.verified.txt#L35)
If there is no numbers as per the [replacement logic here](https://github.com/DataDog/dd-trace-dotnet/blob/a064bf56ef3b9acaccf1d246d43501de1f701faa/tracer/test/Datadog.Trace.Security.IntegrationTests/VerifyScrubber.cs#L57), this wouldnt print <SessionFp> but just ---